solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -63,7 +67,29 @@ pub mod solana_ibc {
                 value: msg.value,
             };
             let res = ibc::core::MsgEnvelope::try_from(msg).and_then(|msg| {
-                ibc::core::dispatch(&mut store, &mut router, msg)
+                let result =
+                    ibc::core::dispatch(&mut store, &mut router, msg.clone());
+                match msg {
+                    MsgEnvelope::Packet(packet) => {
+                        // store the packet if not exists
+                        let mut inner_store = store.0.borrow_mut();
+                        let serialized_packet = borsh::to_vec(&packet).unwrap();
+                        // Find if the packet already exists
+                        match inner_store
+                            .packets
+                            .0
+                            .iter()
+                            .find(|&pack| pack == &serialized_packet)
+                        {
+                            Some(_) => (),
+                            None => {
+                                inner_store.packets.0.push(serialized_packet)
+                            }
+                        }
+                    }
+                    _ => (),
+                }
+                result
```

Comment:
Could we instead generate a new PDA on demand? Something with seed being `["packet", client_id, connection_id, port_id, channel_id, sequence]`.

Also, how are we deleting packets from the list?

Reply:
well we are not deleting the packets from the list. But i think we can delete them once they are acknowledged?

Reply:
For PDA in demand, i think we can